### PR TITLE
[ENH] Classifier Notebook sklearn Examples

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
Adds sklearn model selection and cross-validation examples to the classification notebook (#2116, #2132).

Cleans up the notebook a bit, i.e. faster parameters, typos.

### CODE REVIEW

**Review by fkiraly — DISMISSED** [review]
question: why are we replacing `tsfresh` by `catch22`? The former is more popular afaik.

**Comment by MatthewMiddlehurst** [comment]
It is faster and less verbose, it is minor, so I don't mind reverting.

**Comment by MatthewMiddlehurst** [comment]
unrelated, but TapNet is breaking the macOS test and apparently takes 10 minutes per test

**Comment by fkiraly** [comment]
maybe add `tsfresh` as another example and say fit/predict works the same, just don't run it in the notebook

**Comment by fkiraly** [comment]
> unrelated, but TapNet is breaking the macOS test and apparently takes 10 minutes per test

can you please report? This passed in the PR

**Comment by fkiraly** [comment]
other question: is it correct that the sklearn wrapped classifiers won't accept the pandas formats, or cannot be used themselves in pipelines?

Should we hence not have sktime grid search compositors for classification?

**Comment by MatthewMiddlehurst** [comment]
Unsure how it would fit into a pipeline, output should be the same as normal sklearn estimators

File: examples/02_classification.ipynb [hunk 1]
```diff
@@ -258,22 +235,13 @@
...
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {
-    "collapsed": false
-   },
-   "source": [
+    "## Building Classifiers\n",
+    "\n",
     "We demonstrate the simplest use cases for classifiers and demonstrate how to build bespoke pipelines for time series classification.\n",
```
Comment (TonyBagnall) [inline examples/02_classification.ipynb:240]: change one demonstrate to show

File: examples/02_classification.ipynb [hunk 2]
```diff
@@ -317,9 +285,8 @@
     "collapsed": false
    },
    "source": [
-    "`sktime` contains the state of the art in time series classifiers in the package\n",
-    "classification. These are grouped based on their representation. An accurate and\n",
-    "relatively fast classifier is called [ROCKET](https://arxiv.org/abs/1910.13051)"
+    "`sktime` contains the state of the art in time series classifiers in the package classification. These are grouped based on their representation. An accurate and\n",
```
Comment (TonyBagnall) [inline examples/02_classification.ipynb:288]: dont need this newline after an accurate and

File: examples/02_classification.ipynb [hunk 3]
```diff
@@ -377,16 +342,13 @@
...
    },
    "source": [
-    "Most classifiers in `sktime` involve some degree of transformation. The simplest form\n",
-    "is simply consisting of a pipeline of transformation (aka \"feature extraction\") followed by an `sklearn` classifier.\n",
+    "## Creating Pipelines\n",
     "\n",
-    "The `sktime` `make_pipeline` utility allows to combine transformers and classifiers into a simple pipeline.\n",
-    "The classifier pipelined can be an `sktime` time series classifier, or an `sklearn` tabular classifier.\n",
-    "If an `sklearn` classifier, the time series are formatted as (instance, time index) formatted 2D array before being passed to the `sklearn` classifier.\n",
+    "Most classifiers in `sktime` involve some degree of transformation. The simplest form is simply consisting of a pipeline of transformation (aka \"feature extraction\") followed by an `sklearn` classifier.\n",
```
Comment (TonyBagnall) [inline examples/02_classification.ipynb:347]: The simplest form consists of a

File: examples/02_classification.ipynb [hunk 4]
```diff
@@ -377,16 +342,13 @@
...
     "\n",
-    "In the following example, we use the [tsfresh](https://tsfresh.readthedocs.io/en/latest/) feature extractor to\n",
-    "extract features which are then used in a (tabular, `sklearn`) random forest classifier.\n",
-    "This can be done with the `sktime`'s `make_pipeline` utility as follows:"
+    "The `sktime` `make_pipeline` utility allows to combine transformers and classifiers into a simple pipeline. The classifier pipelined can be an `sktime` time series classifier, or an `sklearn` tabular classifier. If an `sklearn` classifier, the time series are formatted as (instance, time index) formatted 2D array before being passed to the `sklearn` classifier.\n",
```
Comment (TonyBagnall) [inline examples/02_classification.ipynb:349]: utility allows the combination of transformers and classifiers

**Comment by TonyBagnall** [comment]
existing typo "utiltiy" to correct ""Under the hood, `sktime`'s `make_pipeline` utiltiy dispatches to right pipeline"

File: examples/02_classification.ipynb [hunk 5]
```diff
@@ -550,17 +589,12 @@
     }
    },
    "source": [
-    "sktime offers two other ways of solving multivariate time series classification \n",
-    "problems:\n",
+    "`sktime` offers two other ways of solving multivariate time series classification problems:\n",
```
Comment (TonyBagnall) [inline examples/02_classification.ipynb:592]: "solving multivariate time series problems" to "building estimators for multivariate time series problems"

File: examples/02_classification.ipynb [hunk 6]
```diff
@@ -588,8 +622,7 @@
     "collapsed": false
    },
    "source": [
-    "We can also fit one classifier for each time series column and then aggregated their \n",
```
Comment (TonyBagnall) [inline examples/02_classification.ipynb:591]: and then aggregated -> and then aggregate

File: examples/02_classification.ipynb [hunk 7]
```diff
@@ -626,54 +659,39 @@
...
     "\n",
-    "#### The RocketClassifier \n",
-    "is based on a pipeline combination of the ROCKET transformation (transformations\n",
-    ".panel.rocket) and the sklearn RidgeClassifierCV classifier. The RocketClassifier is \n",
-    "configurable to use variants minirocket and multirocket. ROCKET is based on \n",
-    "generating random convolutions. A large number are generated then the classifier \n",
-    "performs a feature selection.\n",
+    "#### KNeighborsTimeSeriesClassifier\n",
+    "\n",
+    "One nearest neighbour (1-NN) classification with Dynamic Time Warping (DTW) is one ofthe oldest TSC approaches, and is commonly used as a performance benchmark.\n",
```
Comment (TonyBagnall) [inline examples/02_classification.ipynb:664]: one of the oldest

**Review by TonyBagnall — CHANGES_REQUESTED** [review]

**Review by TonyBagnall — APPROVED** [review]

**Comment by TonyBagnall** [comment]
contributes to https://github.com/sktime/sktime/issues/1549

**Comment by TonyBagnall** [comment]
> maybe add `tsfresh` as another example and say fit/predict works the same, just don't run it in the notebook

@fkiraly I think this change has been made, could you unblock or let us know if there are other changes you want please? thanks.

**Pull request merged**